FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,45 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case .success(let response):
+
+				guard response.1.statusCode == 200 else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				guard !response.0.isEmpty else {
+					completion(.success([]))
+					return
+				}
+
+				do {
+					let feedLoaderItems = try JSONDecoder().decode(FeedItemAPI.self, from: response.0).items
+					guard let feedImage = self?.feedImagesFactory(feedLoaderItems) else {
+						completion(.success([]))
+						return
+					}
+					completion(.success(feedImage))
+				} catch {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
The error that is thrown by `JSONDecoder.decode` is not used thus there's no need to catch it. To simplify the structure optional `try?` can be used.

Also to follow up on the comment on line 23 optional `self` can be removed

```suggestion
				guard let feedLoaderItems = try? JSONDecoder().decode(FeedItemAPI.self, from: response.0).items else {
					completion(.failure(Error.invalidData))
					return 
				}

				let feedImage = self.feedImagesFactory(feedLoaderItems)
				completion(.success(feedImage))
```